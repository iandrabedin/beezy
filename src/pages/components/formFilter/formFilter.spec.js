import React from "react";
import { cleanup, render } from "@testing-library/react";
import FormFilter from "./formFilter";

afterEach(cleanup);

const props = {
  searchTerm: "title",
  sortTerm: "rating-descending",
  handleSortingChange: jest.fn(),
  handleSearchChange: jest.fn()
};

describe("FormFilter", () => {
  it("should have the title filter", () => {
    const { getByLabelText } = render(<FormFilter {...props} />);
    expect(getByLabelText("Title")).toBeDefined();
  });

  it("should have the sorting select", () => {
    const { getByLabelText } = render(<FormFilter {...props} />);
    expect(getByLabelText("Sort by")).toBeDefined();
  });

  it("should have all the options for select", () => {
    const { getByText } = render(<FormFilter {...props} />);
    expect(getByText("Title Descending")).toBeDefined();
    expect(getByText("Title Ascending")).toBeDefined();
    expect(getByText("Rating Descending")).toBeDefined();
    expect(getByText("Rating Ascending")).toBeDefined();
  });

  it("should render as expected", () => {
    const { asFragment } = render(<FormFilter {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
