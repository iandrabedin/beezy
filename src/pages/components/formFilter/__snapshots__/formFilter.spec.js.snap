// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFilter should render as expected 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <div
      class="form-input"
    >
      <label
        for="filterTitle"
      >
        Title
      </label>
      <input
        id="filterTitle"
        name="filterTitle"
        placeholder="Filter by title..."
        type="text"
        value="title"
      />
    </div>
    <div
      class="form-input"
    >
      <label
        for="sorting"
      >
        Sort by
      </label>
      <select
        id="sorting"
        name="sorting"
      >
        <option
          disabled=""
          value="default"
        >
          Choose here
        </option>
        <option
          value="title-descending"
        >
          Title Descending
        </option>
        <option
          value="title-ascending"
        >
          Title Ascending
        </option>
        <option
          value="rating-descending"
        >
          Rating Descending
        </option>
        <option
          value="rating-ascending"
        >
          Rating Ascending
        </option>
      </select>
    </div>
  </form>
</DocumentFragment>
`;
